pkg/types/vsphere/validation/platform.go

```diff
@@ -221,3 +223,100 @@ func validateLoadBalancer(lbType configv1.PlatformLoadBalancerType) bool {
+		}
+
+		// Check networking
+		if host.NetworkDevice == nil {
+			allErrs = append(allErrs, field.Required(fldPath.Child("networkDevice"), "must specify networkDevice configuration"))
+		} else {
+			allErrs = append(allErrs, validateHostNetworking(host.NetworkDevice, fldPath)...)
+		}
+	}
+
+	return allErrs
+}
+
+// validateHostRole returns error if the host role is invalid
+func validateHostRole(host *vsphere.Host, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	validHostRoles := sets.NewString(vsphere.BootstrapRole, vsphere.ControlPlaneRole, vsphere.WorkerRole)
+	if !validHostRoles.Has(host.Role) {
+		errMsg := fmt.Sprintf("role must be one of %v", validHostRoles.List())
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("role"), host.Role, errMsg))
+	}
+	return allErrs
+}
+
+// validateHostFailureDomain returns error if the FailureDomain is not found
+func validateHostFailureDomain(host *vsphere.Host, fds []vsphere.FailureDomain, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	fdFound := false
+	for _, domain := range fds {
+		if domain.Name == host.FailureDomain {
+			fdFound = true
+			break
+		}
+	}
+	if !fdFound {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("failureDomain"), host.FailureDomain, "failure domain not found"))
+	}
+	return allErrs
+}
+
+// validateHostNetworking checks all fields related to networking for a host.  If any errors are found, they will
+// be returned (invalid IP, IP required)
+func validateHostNetworking(network *vsphere.NetworkDeviceSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// Check ip addresses
+	if len(network.IPAddrs) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("ipAddrs"), "must specify a IP"))
+	}
+	for _, ip := range network.IPAddrs {
+		allErrs = append(allErrs, validateIP(ip, false, fldPath.Child("ipAddrs"))...)
+	}
+
+	// Check nameservers
+	if len(network.Nameservers) > 3 {
+		allErrs = append(allErrs, field.TooMany(fldPath.Child("nameservers"), len(network.Nameservers), 3))
+	}
+	for _, nameserver := range network.Nameservers {
+		allErrs = append(allErrs, validateIP(nameserver, false, fldPath.Child("nameservers"))...)
+	}
+
+	/// Check gateway4
+	allErrs = append(allErrs, validateIP(network.Gateway4, false, fldPath.Child("gateway4"))...)
+
+	// Check gateway6
+	allErrs = append(allErrs, validateIP(network.Gateway6, false, fldPath.Child("gateway6"))...)
+
+	return allErrs
+}
+
+// validateIP checks IP value to see if it is valid.  If IP is required, an error will be returned if
+// the IP is not specified.
+func validateIP(ip string, req bool, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if ip == "" && req {
+		allErrs = append(allErrs, field.Required(fldPath, "must specify a IP"))
+	} else if ip != "" {
+		if valErr := validate.IP(ip); valErr != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, ip, valErr.Error()))
+		}
```

Comment:
The [CAPV API spec](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/troubleshooting.md#a-static-ip-address-must-include-the-segment-length) is ambiguous and yesterday I was questioning how the CIDR was communicated.  Turns out I we have to [provide the CIDR](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/docs/troubleshooting.md#a-static-ip-address-must-include-the-segment-length) as well.

Reply:
Unit tested with the addition of this function for the validation of the host IP address:
`
// validateIPWithCidr checks IP/CIDR value to see if it is valid.  If IP is required, an error will be returned if
// the IP is not specified.
func validateIPWithCidr(ip string, req bool, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}
	if ip == "" && req {
		allErrs = append(allErrs, field.Required(fldPath, "must specify a IP address with CIDR"))
	} else if ip != "" {
		if _, _, valErr := net.ParseCIDR(ip); valErr != nil {
			allErrs = append(allErrs, field.Invalid(fldPath, ip, valErr.Error()))
		}
	}
	return allErrs
}
`

Reply:
Ok.  Updated API / functions to accept CIDR.